Including liquibase in project (added 2 tables)

Fenydude opened the pull request.

ShirshovAlex committed:
Entity part two (#3)

* Internship entity

* Internship entity

Maksim-maevsky committed:
created entity Form and Interview (#4)

ShirshovAlex committed:
entity part three (#5)

Co-authored-by: Olga <olgaaevdokim@gmail.com>

Maksim-maevsky committed:
uncomented variables

Maksim-maevsky committed:
changed LocalDate instead Date , created InternshipLocation

Maksim-maevsky committed:
added InternshipLocation to Internship

Maksim-maevsky committed:
added UUID? and changed variable PATH in DockerFile like path

Maksim-maevsky committed:
changed int

Maksim-maevsky committed:
added timeSlot

ShirshovAlex committed:
change enum timeslot

ShirshovAlex committed:
change enum timeslot

Maksim-maevsky committed:
added TimeForCall

Maksim-maevsky committed:
removed Role_ prefix for user status, removed package team_four, removed Internship word from name of date variable.

Maksim-maevsky committed:
removed prefix Level for variables of Enum EnglishLevel, create property variables for versions in pom.xml etc

Maksim-maevsky committed:
changed name of class InternshipLocation? now it is Location.
create txt file doc.txt / removed  WAITING, GOT_JOB_OFFER, DIDNT_GOT_JOB_OFFER from RoleStatus and copy in doc.txt,

Maksim-maevsky committed:
InternshipFormat instead FormatOfInternship

Fenydude committed:
added yaml & test code

Fenydude committed:
merging with development

Fenydude committed:
fixed conflict in enam

OlgaEvdokimova committed:
add db.chengelog-internship.yaml, internship.sql

OlgaEvdokimova committed:
Merge branch 'liquibase' of https://github.com/viktarkolbik/backend_team_four into liquibase

ShirshovAlex committed:
Shirshov liquibase (#8)

* add form table

* Delete db.changelog-item.yaml

* add form table

* add changeSet with sqlPath

* add uuid-ossp in sqlFile

Fenydude committed:
tested liquibase

Fenydude committed:
make a sprint folder && move changelog files in sprint folder

Fenydude committed:
move version of dependencies to properties

mortal-gpt commented on doc.txt:
```diff
@@ -0,0 +1,6 @@
+
+    1. Forms statuses for future implementation
```
Move documentation to README.MD

ShirshovAlex replied:
done

mortal-gpt commented on pom.xml:
```diff
@@ -56,8 +56,29 @@
+            <version>${postgresql.version}</version>
+        </dependency>
+
+        <!-- https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-data-jpa -->
```
Looks like you copied dependency with this comment, remove it from all places please

ShirshovAlex replied:
done

mortal-gpt commented on src/main/resources/application.yaml:
```diff
@@ -0,0 +1,13 @@
+    username: postgres
+  liquibase:
+    enabled: true
+    change-log : classpath:db/changelog/db.changelog-master.yaml
```
Is this the same as default value? If yes, remove please.

ShirshovAlex replied:
done

mortal-gpt commented on src/main/resources/db/changelog/v.0.1/db.changelog-form.yaml:
```diff
@@ -0,0 +1,62 @@
+databaseChangeLog:
+  - changeSet:
+      id: 31032021-add-form-table
+      author: alexei
```
put firstName_lastName please. (In real project we use email, but here we expect that your email will be less informative).

Apply it to other places

ShirshovAlex replied:
done

kzaretskaya replied:
@ShirshovAlex @OlgaEvdokimova please be attentive, you left changeSet with unchanged `author`.

ShirshovAlex replied:
100% done

mortal-gpt commented on src/main/resources/db/changelog/v.0.1/db.changelog-form.yaml:
```diff
@@ -0,0 +1,62 @@
+                  constraints:
+                    nullable: false
+              - column:
+                  name: fm_midlename
```
```suggestion
                  name: fm_middlename
```

ShirshovAlex replied:
done

mortal-gpt commented on src/main/resources/db/changelog/v.0.1/db.changelog-form.yaml:
```diff
@@ -0,0 +1,62 @@
+                    unique: true
+                    primaryKey: true
+              - column:
+                  name: fm_name
```
```suggestion
                  name: fm_firstname
```

ShirshovAlex replied:
done

mortal-gpt commented on src/main/resources/db/changelog/v.0.1/db.changelog-form.yaml:
```diff
@@ -0,0 +1,62 @@
+      changes:
+        dbms: postgresql
+        sqlFile:
+          path: db/changelog/v.0.1/sqlFile/insert_form.sql
```
As far as I understand, you want to fill the table with test data. 
To my mind it's better not to do this with liquibase, you can populate DB using Hibernate functionality. So, drop this changset, please.

ShirshovAlex replied:
done

ShirshovAlex committed:
moved documentation to README.MD
removed comments from pom.xml
removed change-log from application.yaml
changed author names in db.files

ShirshovAlex committed:
Merge remote-tracking branch 'origin/liquibase' into liquibase

# Conflicts:
#	pom.xml

ShirshovAlex committed:
Delete doc.txt

ShirshovAlex committed:
Merge remote-tracking branch 'origin/liquibase' into liquibase

ShirshovAlex committed:
dropped change-set v.0.1
deleted sql folder

ShirshovAlex committed:
Delete src/main/resources/db/changelog/v.0.1/sqlFile directory

ShirshovAlex committed:
return sqlFile folder with inserts

kzaretskaya commented on src/main/resources/db/changelog/v.0.1/db.changelog-internship.yaml:
```diff
@@ -0,0 +1,51 @@
+                  name: inship_tech_skills
+                  type: VARCHAR(255)
+                  constraints:
+                    nullable: false
```
Remove please non nullable constraint for fields `inship_tech_skills` `inship_description` and `inship_requirements`.
As far as I remember Zhanna said that internship can be created without additional information.
If in future we recognize that some field should be required appropriate constraint can be added.

ShirshovAlex replied:
done

kzaretskaya commented on src/main/resources/db/changelog/v.0.1/db.changelog-form.yaml:
```diff
@@ -0,0 +1,55 @@
+                  type: VARCHAR(255)
+              - column:
+                  name: fm_education
+                  type: VARCHAR(100)
```
may be better VARCHAR(255)?

ShirshovAlex replied:
done

kzaretskaya commented on src/main/resources/db/changelog/db.changelog-master.yaml:
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/v.0.1/db.changelog-master-v.0.1.yaml
```
Path indicates what version is used, postfix is excessive.

```suggestion
      file: db/changelog/v.0.1/db.changelog-master.yaml
```

ShirshovAlex replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/dto/UserDTO.java:
```diff
@@ -25,5 +24,4 @@
     private UserRole userRole;
     private List<InternshipDTO> listOfInternships;
     private List<LocalDateTime> freeInterviewDates;
-
```
Thank you for keeping file clear! 🥇

kzaretskaya commented on pom.xml:
```diff
@@ -49,13 +44,29 @@
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>42.2.19</version>
```
Suggestion to keep all versions in one place - properties. Please, move it. The same for `liquibase-core`

ShirshovAlex replied:
done

kzaretskaya commented on src/main/resources/db/changelog/v.0.1/db.changelog-master-v.0.1.yaml:
```diff
@@ -0,0 +1,5 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/v.0.1/db.changelog-internship.yaml
```
Enough to add here only fileName because `db.changelog-master-v.0.1.yaml ` is in the same path.

ShirshovAlex replied:
we changed as u said but it doesnt work. Only with full filePath

kzaretskaya replied:
Sorry, I missed tag `relativeToChangelogFile`. try with it. It will look like
databaseChangeLog:
  - include:
      file: db.changelog-internship.yaml
      relativeToChangelogFile: true

ShirshovAlex replied:
already done

ShirshovAlex committed:
removed nullable constraints
change file name db.changelog-master
moved version properties in pom.xml

Fenydude committed:
set relative path to file

Fenydude committed:
set relative path to file
